src/python/pants/engine/target.py

```diff
@@ -379,6 +379,7 @@ class Target:
     field_values: FrozenDict[type[Field], Field]
     residence_dir: str
     name_explicitly_set: bool
+    build_file_source: str
```

Comment:
@stuhood do we have any benchmarks or so for the impact of adding more fields to targets?

I recall that fields don't have the target's address as a member attribute by default for perf reasons.. guess fields are at a larger scale, but for large repos, I guess targets might have an impact too.. ?